### ISSUE TITLE
Bug in addtorque example, compute chunk/atom after atom exchange

### ISSUE THREAD

**Issue by rbberger** [lammps/lammps#390]
I've been chasing down a bug in the addtorque example which has been broken in our [regression testing](https://ci.lammps.org/job/lammps/job/master/job/regression/54/testReport/examples.USER.misc/addtorque/addtorque/) for some time now. It crashes with a segmentation fault in some cases (e.g., running with 8 processors). I found the reason for this is to be uninitialized memory in the `ichunk` array of compute_chunk_atom.

```bash
# part of examples/USER/misc/addtorque/in.addtorque
...
# equilibration w/o torque added
run 1000 post no

# start spinning the ball. rotation around z should increase and Erot_z should grow
fix 4 ball addtorque 0.0 0.0 200.0
run 10000 upto  post no   # <---- crash happens at the beginning of this run
```

Going deeper, it turns out that number of atoms on each processor changes between the two runs. Some atoms are exchanged, for whatever reason. However, since `compute_ichunk()` was executed at timestep 1000 of the previous run, the ichunk array is not updated due to line [699](https://github.com/lammps/lammps/blame/master/src/compute_chunk_atom.cpp#L699) in compute_chunk_atom.cpp. This seems to be the reason why memory is not initialized.

TL;DR: the `ichunk` array isn't recomputed although atoms have moved to different processors at the beginning of a new run

**Comment by akohlmey** [comment]
Does the following patch restore the expected behavior?
```
diff --git a/src/compute_chunk_atom.cpp b/src/compute_chunk_atom.cpp
index 925c5fb..ff7eccb 100644
--- a/src/compute_chunk_atom.cpp
+++ b/src/compute_chunk_atom.cpp
@@ -614,6 +614,7 @@ void ComputeChunkAtom::init()
 
 void ComputeChunkAtom::setup()
 {
+  invoked_ichunk = -1;
   if (nchunkflag == ONCE) setup_chunks();
   if (idsflag == ONCE) compute_ichunk();
 }
```

**Comment by rbberger** [comment]
Yup, that does the trick.
